frontend/js/src/explore/artist-similarity/Data.tsx

```diff
@@ -0,0 +1,244 @@
+    Array<ArtistType>
+  >([]);
+  const [mainArtist, setMainArtist] = useState<ArtistType>();
+  const [similarArtistsLimit, setSimilarArtistsLimit] = useState(
+    SIMILAR_ARTISTS_LIMIT_VALUE
+  );
+  const [colors, setColors] = useState([color1, color2]);
+  const [artistMBID, setArtistMBID] = useState(ARTIST_MBID);
+
+  const scoreList: Array<number> = [];
+
+  const processData = useCallback(
+    (dataResponse: ApiResponseType): void => {
+      // Type guard for dataset response
+      const isDatasetResponse = (
+        response: MarkupResponseType | DatasetResponseType
+      ): response is DatasetResponseType => {
+        return response.type === "dataset";
+      };
+      // Get the datasets out of the API response
+      const artistsData = dataResponse.filter(isDatasetResponse);
+      if (artistsData.length) {
+        // Get the main artist from the first dataset
+        setMainArtist(artistsData[0].data[0]);
+        // Get the similar artists from the second dataset
+        const similarArtistsResponse = artistsData[1];
+        if (similarArtistsResponse?.data?.length) {
+          setCompleteSimilarArtistsList(similarArtistsResponse.data);
+          setSimilarArtistsList(
+            similarArtistsResponse.data.slice(0, similarArtistsLimit)
+          );
+        }
+        // In case no similar artists are found
+        else {
+          setSimilarArtistsList([]);
+        }
+      }
+      setColors([tinycolor.mix(color1, color2, COLOR_MIX_WEIGHT), color2]);
+    },
+    [artistMBID]
+  );
+
+  const fetchData = useCallback(
+    async (artist_mbid: string): Promise<void> => {
+      try {
+        const response = await fetch(BASE_URL + artist_mbid);
+        const data = await response.json();
+        processData(data);
+      } catch (error) {
+        // Error message goes here.
+        alert(
+          "Something went wrong while loading information, please try again"
+        );
+      }
+    },
+    [processData]
+  );
+
+  // Update the graph when either artistMBID or similarArtistsLimit changes
+  useEffect(() => {
+    fetchData(artistMBID);
+  }, [artistMBID, fetchData]);
+  // Update the graph when limit changes by only changing the data and not making a new request to server
+  useEffect(() => {
+    const newSimilarArtistsList = completeSimilarArtistsList.slice(
+      0,
+      similarArtistsLimit
+    );
+    setSimilarArtistsList(newSimilarArtistsList);
+  }, [similarArtistsLimit]);
+
+  // Calculating minScore for normalization which is always the last element of the array (because it's sorted)
+  let minScore = similarArtistsList?.[similarArtistsLimit - 1]?.score ?? 0;
+  minScore = Math.sqrt(minScore);
+
+  // Transforming the data into the format required by the graph
+  let transformedArtists: GraphDataType = {
+    nodes: [],
+    links: [],
+  };
+  // Checking if mainArtist is defined
+  if (mainArtist) {
+    transformedArtists = {
+      nodes: [mainArtist, ...similarArtistsList].map(
```

Comment:
This should probably be separated into a separate function.

Reply:
It seems like there is no easy way to separate the function. I have added the comment for it if you can please take a closer look.

Reply:
Sure, I have pushed an updated version.